src/snyk/common/languageServer/languageServer.ts

```diff
@@ -88,34 +88,43 @@ export class LanguageServer implements ILanguageServer {
     // Create the language client and start the client.
     this.client = this.languageClientAdapter.create('Snyk LS', SNYK_LANGUAGE_SERVER_NAME, serverOptions, clientOptions);
 
-    this.client.onNotification(SNYK_HAS_AUTHENTICATED, ({ token }: { token: string }) => {
-      this.authenticationService.updateToken(token).catch((error: Error) => {
-        ErrorHandler.handle(error, this.logger, error.message);
-      });
-    });
-
-    this.client.onNotification(SNYK_CLI_PATH, ({ cliPath }: { cliPath: string }) => {
-      if (!cliPath) {
-        ErrorHandler.handle(
-          new Error("CLI path wasn't provided by language server on $/snyk.isAvailableCli notification"),
-          this.logger,
-          "CLI path wasn't provided by language server on notification",
-        );
-        return;
-      }
-
-      void this.configuration
-        .setCliPath(cliPath)
-        .then(() => {
-          this.cliReady$.next(cliPath);
-        })
-        .catch((error: Error) => {
-          ErrorHandler.handle(error, this.logger, error.message);
+    this.client
```

Comment:
Just a very minor point, but can this block be promisified with `util.promifiy` to avoid mixing promise chains and async/await?

Reply:
It's possible to await the promise here without promisify, but we don't want to block here, because LanguageClient won't start otherwise.

Reply:
One can argue we can await after `client.start()`, however I am cautious of race condition in case we're too late to register listeners, if we await `onReady` after the start itself.